xonsh/xoreutils/_which.py

```diff
@@ -191,21 +192,21 @@ def whichgen(command, path=None, verbose=0, exts=None):
     # Windows has the concept of a list of extensions (PATHEXT env var).
     if sys.platform.startswith("win"):
         if exts is None:
-            exts = os.environ.get("PATHEXT", "").split(os.pathsep)
+            exts = builtins.__xonsh_env__['PATHEXT']
```

Comment:
Are we talking about different things? I was not talking about `exts = os.environ.get("PATHEXT", "").split(os.pathsep)` We don't use that. `exts` are passed in thought the environments. It is the way `PATH` is passed in by chaning the os.environ temporarily.  I think we can wait with such a  change